File: front/lib/dto/gathering_hotList_dto.dart
import 'package:front/screen/gathering_screen/model/gathering_card_model.dart';

class GatheringHotlistDto {
  final GatheringMeetingModel meetingInfo;

  final int maxParticipants;
  final String location;
  final String categoryName;
  final DateTime meetingDate;

  const GatheringHotlistDto({
    required this.meetingInfo,
    required this.maxParticipants,
    required this.location,
    required this.categoryName,
    required this.meetingDate,
  });

  factory GatheringHotlistDto.fromJson(Map<String, dynamic> json) {
    return GatheringHotlistDto(
      meetingInfo: GatheringMeetingModel(
        title: json['roomName'] ?? '',
        content: json['roomDesc'] ?? '',
        imageUrl: json['roomImg'] ?? '',
        currentParticipants: json['usersIdx'] ?? 0,
      ),
      maxParticipants: json['max'] ?? 0,
      location: json['meetPlace'] ?? '',
      categoryName: json['roomType'] ?? '',
      meetingDate: json['meetDate'] != null
      ? DateTime.parse(json['meetDate'] as String)
      : DateTime.now(),
    );
  }
}

File: backend/src/main/java/com/teamj/dto/gathering_dto/GatheringHotListDto.java
package com.teamj.dto.gathering_dto;

import java.time.LocalDateTime;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class GatheringHotListDto {
    private Long roomIdx;
    private String roomName;
    private String roomType;
    private LocalDateTime createdAt;
    private LocalDateTime meetDate;
    private String meetPlace;
    private int max;
    private String roomImg;
    private String roomDesc;
}